packages/fxa-settings/src/components/Settings/PageRecoveryKeyCreate/index.test.tsx

```diff
@@ -18,30 +26,83 @@ jest.mock('@reach/router', () => ({
   useNavigate: () => mockNavigate,
 }));
 
+jest.mock('base32-encode', () =>
+  jest.fn().mockReturnValue('00000000000000000000000000000000')
+);
+
+window.URL.createObjectURL = jest.fn();
+
+const accountWithSuccess = {
+  ...MOCK_ACCOUNT,
+  createRecoveryKey: jest.fn().mockResolvedValue(new Uint8Array(20)),
+} as unknown as Account;
+
+const renderPageWithContext = (account: Account) => {
+  renderWithRouter(
+    <AppContext.Provider value={mockAppContext({ account })}>
+      <PageRecoveryKeyCreate />
+    </AppContext.Provider>
+  );
+};
+
 describe('PageRecoveryKeyCreate', () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
-  it('renders as expected', () => {
-    // Content will be updated as views are completed.
-    render(<PageRecoveryKeyCreate />);
+  it('renders initial flow page as expected', () => {
+    renderPageWithContext(accountWithSuccess);
     expect(
       screen.getByText(
         'Create an account recovery key in case you forget your password'
       )
     ).toBeInTheDocument();
   });
-  it('shifts views when the user clicks through the flow steps', () => {
-    render(<PageRecoveryKeyCreate />);
+  it('shifts views when the user clicks through the flow steps', async () => {
+    renderPageWithContext(accountWithSuccess);
+
+    // Go to page 2
+    const flowPage1Button = screen.getByRole('button', {
+      name: 'Start creating your recovery key',
+    });
+    await act(async () => {
+      await fireEvent.click(flowPage1Button);
+    });
     expect(
-      screen.getByText(
-        'Create an account recovery key in case you forget your password'
-      )
+      screen.getByRole('heading', {
+        name: 'Enter your password again to get started',
+      })
+    ).toBeInTheDocument();
+
+    // Go to page 3
+    const flowPage2Button = screen.getByRole('button', {
+      name: 'Create recovery key',
+    });
+    // input a password to enable form submission
+    const inputPassword = typeByLabelText('Enter your password');
+    await act(async () => {
+      inputPassword('anypassword');
+    });
+    await act(async () => {
+      await fireEvent.click(flowPage2Button);
```

Comment:
You shouldn't need `await` for `fireEvent`, it's in a couple spots here.

Reply:
Ah yes, so right! I need to await the result, not the event.